discovery_engine/lib/src/domain/engine/engine.dart

```diff
@@ -72,6 +72,36 @@ abstract class Engine {
     Set<Source> trusted,
   );
 
+  /// Returns the excluded sources.
+  Future<Set<Source>> getExcludedSources();
+
+  /// Returns the trusted sources.
+  Future<Set<Source>> getTrustedSources();
+
+  /// Adds a new excluded source to the current set of excluded sources.
+  Future<void> addExcludedSource(
+    List<SourceReacted> sources,
+    Source excluded,
+  );
+
+  /// Removes a new excluded source to the current set of excluded sources.
```

Comment:
```suggestion
  /// Removes an excluded source from the current set of excluded sources.
```
:art:  also check `removeTrustededSource` and docstrings on the rust side